src/ActionSheet/index.tsx

```diff
@@ -0,0 +1,275 @@
+import * as React from 'react';
+import {
+  Animated,
+  BackHandler,
+  Easing,
+  StyleSheet,
+  TouchableWithoutFeedback,
+  View,
+} from 'react-native';
+import ActionGroup from './ActionGroup';
+import { ActionSheetOptions } from '../types';
+
+type ActionSheetState = {
+  isVisible: boolean;
+  isAnimating: boolean;
+  options: ActionSheetOptions | null;
+  onSelect: ((i: number) => void) | null;
+  overlayOpacity: any;
+  sheetOpacity: any;
+};
+
+type ActionSheetProps = {
+  readonly useNativeDriver: boolean | undefined;
+};
+
+const OPACITY_ANIMATION_IN_TIME = 225;
+const OPACITY_ANIMATION_OUT_TIME = 195;
+const EASING_OUT = Easing.bezier(0.25, 0.46, 0.45, 0.94);
+const EASING_IN = Easing.out(EASING_OUT);
+
+// Has same API as https://facebook.github.io/react-native/docs/actionsheetios.html
+export default class ActionSheet extends React.Component<ActionSheetProps, ActionSheetState> {
+  static defaultProps = {
+    useNativeDriver: true,
+  };
+
+  _actionSheetHeight = 360;
+  _animateOutCallback: (() => void) | null = null;
+
+  state: ActionSheetState = {
+    isVisible: false,
+    isAnimating: false,
+    options: null,
+    onSelect: null,
+    overlayOpacity: new Animated.Value(0),
+    sheetOpacity: new Animated.Value(0),
+  };
+
+  _setActionSheetHeight = ({ nativeEvent }) =>
+    (this._actionSheetHeight = nativeEvent.layout.height);
+
+  render() {
+    const { isVisible, overlayOpacity } = this.state;
+    const overlay = !!isVisible && (
+      <Animated.View
+        style={[
+          styles.overlay,
+          {
+            opacity: overlayOpacity,
+          },
+        ]}
+      />
+    );
+    return (
+      <View
+        style={{
+          flex: 1,
+        }}>
+        {React.Children.only(this.props.children)}
+        {overlay}
+        {!!isVisible && this._renderSheet()}
```

Comment:
can you do `isVisible ? this._renderSheet() : null`?

Reply:
:+1